### PR TITLE
Bug-Fix for subforms / repeatable fields

### PR DESCRIPTION
**Pull request by Lab5-Switzerland**
JS Form Validation for Subforms/Repeatable Fields Bug IE - Here is the Solution

### What needs to be fixed
In IE(11), Fields of a Subform or Repeatable Form dont validate - Seemingly.  
Truth is : IE tries to validate the invisible Subform-repeatable<template> part of the subform - which ofc never contains values by its very nature - thus always producing errors

### Why this should be fixed
Because Client side form validation is essential. The bug described just exists, because repeatable fields are relatively new and this issue hasn't been addressed yet.
Also the soloution is beautifully simple and implemented in a minute.

### How would you fix it
Add the suggested line ( see below ) in to the File 
**/media/system/js/validate-uncompressed.js , circa line 129**, right after 
if(jQuery(fields[i]).hasClass('novalidate')) { continue; }
**ADD** the following line / code :
**if(jQuery(fields[i]).parents().is("template") ) { continue; }**

Explanation : 
If a field is directly of indirectly encapsulated into a <template> tag, it is part of the subform's/repeatable-field's template (/reserve) part, BUT NOT the actual form in use - and thus is NOT to be treated as the part of the form in actual use. Thus, if the validator sees the a given field has as direct or indirect/farther parent a <template> element, the validator knows that this field is not part of the form and will skipp validation on this field. Viola. 
Simple and beautiful solution.
( Reation of the validator is analogue to the line before, where it makes the validator skipp the field, it it sees a "novalidate" class attached ).

Ofc, the soloution must be transferred into the compressed version ( validate.js ) too.

### Side Effects expected
None.

Pull Request for Issue #24829

### Summary of Changes
3 Lines of Code

### Testing Instructions
Create any Subform. Some of the Fields in the Subform need to have validation required. 
( e.g. add a field field of type="email", required="true", validate="email" )
Test in IE(11) by filling the form and trying to submit the form.
You'll get a form error, althou you have entered all fields correctly.

### Expected result
Expected result in form with unfixed file (validate.js) would be, that you can simply submit the form propperly.

### Actual result
Actual result in form with unfixed file (validate.js) is, that you get error messages, althou you seemingly entered everything correctly.

### CODE REVIEW

File: media/system/js/validate-uncompressed.js [hunk 1]
```diff
@@ -127,6 +127,10 @@ var JFormValidator = function() {
  	 		if(jQuery(fields[i]).hasClass('novalidate')) {
  	 			continue;
  	 		}
+			// If field is part of a subform-repeatable template, it's not supposed to be validated.
+ 	 		if(jQuery(fields[i]).parents().is("template") ) {
```
Comment (Quy) [inline media/system/js/validate-uncompressed.js]:
```suggestion
 	 		if(jQuery(fields[i]).parents().is("template")) {
```

**Comment by Fedik** [comment]
2 note:
- it only affect an old EI, so I would not run this hack for every browser, but make some extra check `if (IE and template)`, because of performance 
- instead of `jQuery(fields[i]).parents().is("template")` use `jQuery(fields[i]).closest("template")`

**Comment by Lab5-Switzerland** [comment]
> 
> 2 note:
> 
>     * it only affect an old EI, so I would not run this hack for every browser, but make some extra check `if (IE and template)`, because of performance
> 
>     * instead of `jQuery(fields[i]).parents().is("template")` use `jQuery(fields[i]).closest("template")`

Great ideas !
I already checked ONE of your suggestions ( .closer() ), which i found great - unfortunately that one has a problem (buggy in Joomla!'s jQuery in IE11).

1.a) Are you shure it only affects an old IE ?
2.a) what's the appropriate line for the IE check ? Im not sure about it. If you have it at hand that'd be great. 

2.)
jQuery.closest() WOULD been a great idea, BUT : for some strange reason, it skipps the whole form.
In the end, it seems to be a jQuery based bug anyway in conjunction with IE , evidenced by the behaviour :
- althou structurally / hirarchically, the <template> element lies above the field and BESIDE some ancestor element, closest() will hit on it in IE11 althou it should definitely not ( as far as my understanding of jQuery.closest() goes - it should only hit on elements in DIRECT ancestorial line, not sibblings of ancestors. ( Correct me if i'm wrong )).

**Comment by Fedik** [comment]
sorry, forget what I wrote :)

it more like  a workaround, but not a correct fix, 
need to looks at subform script to fix IE compatibility for this case.

File: media/system/js/validate-uncompressed.js (hunk 1 above)
Comment (Quy) [inline media/system/js/validate-uncompressed.js]:
```suggestion
 	 		if (jQuery(fields[i]).parents().is("template")) {
```

**Comment by Quy** [comment]
Please use the following minifier to submit the compressed version.
https://skalman.github.io/UglifyJS-online/

**Comment by Lab5-Switzerland** [comment]
> Please use the following minifier to submit the compressed version.
> https://skalman.github.io/UglifyJS-online/

Did now. Hope everything went well.

File: media/system/js/validate-uncompressed.js (hunk 1 above)
Comment (Lab5-Switzerland) [reply media/system/js/validate-uncompressed.js]: OK, tidied up - Both lil changes implemented.

**Comment by Fedik** [comment]
please, better test this two:
#24947 for Joomla! 3
#24944 for Joomla! 4

**Comment by Quy** [comment]
Closing in favor of #24947. 

Thanks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24832">issues.joomla.org/tracker/joomla-cms/24832</a>.</sub>

**Pull request closed without merging**